src/app/Dashboard/DraggableRef.tsx

```diff
@@ -0,0 +1,125 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { dashboardCardConfigResizeCardIntent } from '@app/Shared/Redux/Configurations/DashboardConfigSlicer';
+import { gridSpans } from '@patternfly/react-core';
+import _ from 'lodash';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { DashboardCardContext } from './ResizableCard';
+
+export interface DraggableRefProps {
+  dashboardId: number;
+  minimumSpan: gridSpans;
+}
+
+function normalizeAsGridSpans(val: number, min: number, max: number, a: gridSpans, b: gridSpans): gridSpans {
+  if (val < min) val = min;
+  else if (val > max) val = max;
+  const ans = Math.round((b - a) * ((val - min) / (max - min)) + a);
+  return _.clamp(ans, a, b) as gridSpans;
+}
+
+export const DraggableRef: React.FunctionComponent<DraggableRefProps> = (props) => {
+  const dispatch = useDispatch();
+
+  const cardRef = React.useContext(DashboardCardContext);
+  const isResizing = React.useRef<boolean>(false);
+
+  const SMALLEST_CARD_WIDTH = 126;
+
+  const handleOnMouseDown = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    e.stopPropagation();
+    e.preventDefault();
+    document.body.style.setProperty('cursor', 'col-resize');
+    document.addEventListener('mousemove', callbackMouseMove);
+    document.addEventListener('mouseup', callbackMouseUp);
+    isResizing.current = true;
+  };
+
+  const handleMouseMove = (e: MouseEvent) => {
+    const mousePos = e.clientX;
+    e.stopPropagation();
+    if (!isResizing.current) {
+      return;
+    }
+    if (cardRef.current && window.visualViewport) {
+      cardRef.current.scrollIntoView({ behavior: 'smooth', block: 'center', inline: 'end' });
+      const cardRect = cardRef.current.getBoundingClientRect();
+      const minimumCardRight = cardRect.left + SMALLEST_CARD_WIDTH * props.minimumSpan;
+
+      const newSize = mousePos;
+
+      const gridSpan = normalizeAsGridSpans(
+        newSize,
+        minimumCardRight,
+        window.visualViewport.width,
+        props.minimumSpan,
+        12
+      ) as gridSpans;
+
+      dispatch(dashboardCardConfigResizeCardIntent(props.dashboardId, gridSpan));
+    } else {
+      console.error('cardRef.current or window.visualViewport is undefined');
+    }
+  };
+
+  const handleOnMouseUp = () => {
+    if (!isResizing.current) {
+      return;
+    }
+    isResizing.current = false;
+    document.body.style.removeProperty('cursor');
+    document.removeEventListener('mousemove', callbackMouseMove);
+    document.removeEventListener('mouseup', callbackMouseUp);
+  };
+
+  const callbackMouseMove = React.useCallback(handleMouseMove, []);
```

Comment:
Need hooks deps. Probably best to pass directly "handle*" implementation here. Also, need to wrap `handleOnMouseDown` in useCallback too.